src/core/caniuse.js

```diff
@@ -0,0 +1,172 @@
+  }
+}
+
+const DEFAULT_BROWSERS = ["chrome", "firefox", "ie", "edge"];
+const BROWSERS = { // browser name dictionary
+  chrome: "Chrome",
+  firefox: "Firefox",
+  ie: "IE",
+  edge: "Edge",
+  android: "Android",
+  safari: "Safari",
+  opera: "Opera",
+  bb: "Blackberry",
+  and_uc: "UC (Android)",
+  and_ff: "Firefox (Android)",
+};
+
+/**
+ * main canIUse function
+ * @param  {String} key                         which api to look for
+ * @param  {Node} el                          add table after el
+ * @param  {Array:string} [browsers=DEFAULT_BROWSERS] list of browsers to show
+ */
+function canIUse(key, el, browsers = DEFAULT_BROWSERS) {
+  const url = `https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/${key}.json`;
+
+  // use data from localStorage data if valid and render
+  const cached = localStorage.getItem(`caniuse-${key}`);
+  if (cached) {
+    const stats = JSON.parse(cached);
+    const CACHE_DURATION = 24 * 60 * 60 * 1000; // in ms
+    if (new Date() - new Date(stats.$cacheTime) < CACHE_DURATION) {
+      return showData(key, stats, el, browsers);
+    }
+  }
+  // otherwise fetch new data and render
+  const placeholder = createPlaceholder(key, el);
+  const handleResponse = ({ stats }) => showData(key, stats, el, browsers);
+  const handleError = err => showError(err, placeholder);
+  ghFetch(url)
+    .then(handleResponse)
+    .catch(handleError);
+}
+
+/**
+ * creates a placeholder while the API fetches results
+ * @param  {String} key    API name
+ * @param  {Node} parent where to create placeholder
+ * @return {Node}        inserted placeholder
+ */
+function createPlaceholder(key, parent) {
+  const canIUseId = `caniuse-${key}`;
+  const placeholder = hyperHTML`
+    <div class="caniuse" id="${canIUseId}">
+      <dt class="caniuse-title">
+        Can I Use this API? (${key})
+      </dt>
+      <dd>fetching data from caniuse.com... </dl>
+    </div>`;
+  return parent.parentNode.insertBefore(placeholder, parent.nextSibling);
+}
+
+function showError(err, placeholder) {
+  placeholder.style.height = "40px";
+  placeholder.querySelector("dd").innerText = `Error: ${err.message}`;
+}
+
+/**
+ * render the canIUse support table
+ * @param  {String} key      API name
+ * @param  {Object} stats     CanIUse API results
+ * @param  {Node} parent   where to render table
+ * @param  {Array:string} browsers lost of browsers to show support for
+ */
+function showData(key, stats, parent, browsers) {
+  if (typeof browsers === "string") {
+    if (browsers === "ALL") {
+      browsers = Object.keys(stats);
+    } else {
+      pub("warn", "`conf.caniuseBrowsers` is set wrong.");
+      browsers = DEFAULT_BROWSERS;
+    }
+  }
+
+  // utils
+  const canIUseId = `caniuse-${key}`;
+  const permalink = `http://caniuse.com/#feat=${key}`;
+
+  // cache the response
+  stats.$cacheTime = new Date();
+  localStorage.setItem(canIUseId, JSON.stringify(stats));
+
+  const validBrowsers = browsers.filter(b => b in stats);
```

Comment:
I think we should normalize user input. I can see people are going to do:

```JS
browsers: ["Chrome", "FireFox"] 
```

So we should map them to lower case.